### PR TITLE
fix: handle WebSocket errors by default (#1642)

### PR DESCRIPTION
**Pull request by maknapp**
- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

No, not sure how to add tests for this.

### Motivation / Use-Case

Prevents webpack-dev-server from crashing if a proxied WebSocket has errors.

### Breaking Changes

No breaking changes.

### Additional Info
None.

### CODE REVIEW

File: lib/Server.js [hunk 1]
```diff
@@ -242,6 +242,15 @@ class Server {
           ? proxyConfigOrCallback()
           : proxyConfigOrCallback;
 
+      // Catch WebSocket errors by default
+      if (proxyConfig.ws && typeof proxyConfig.onProxyReqWs !== 'function') {
+        proxyConfig.onProxyReqWs = (proxyReq, req, socket) => {
+          socket.on('error', (err) => {
+            this.logger.warn('WebSocket error', err.message);
```
Comment (alexander-akait) [inline lib/Server.js:249]: I think we should add better message here, like `[http-proxy-middleware] Web socket error:` to avoid misleading

**Review by alexander-akait — COMMENTED** [review]
Yes, tests is hard there, maybe you can provide steps to manually tests and I will put in todo adding tests

File: lib/Server.js (hunk 1 above)
Comment (maknapp) [reply lib/Server.js:249]: My output looks like this:
```
<i> [webpack-dev-server] [HPM] Upgrading to WebSocket
<w> [webpack-dev-server] WebSocket error read ECONNRESET
<i> [webpack-dev-server] [HPM] Upgrading to WebSocket
<i> [webpack-dev-server] [HPM] Upgrading to WebSocket
<w> [webpack-dev-server] WebSocket error write after end
<w> [webpack-dev-server] WebSocket error write after end
<i> [webpack-dev-server] [HPM] Upgrading to WebSocket
<w> [webpack-dev-server] WebSocket error write after end
<w> [webpack-dev-server] WebSocket error write after end
```

Are you saying to change it to `this.logger.warn('WebSocket error:', err.message);` to make it:
```
<w> [webpack-dev-server] WebSocket error: write after end
```
or add `[http-proxy-middleware]` also to make it:
```
<w> [webpack-dev-server] [http-proxy-middleware] WebSocket error: write after end
```

**Comment by maknapp** [comment]
I was trying to write tests and then realized that this is for 4.0.0-beta versions and not 3.11.0 which is where I was seeing the issue. I did not realize that master was so far ahead. I tried to reproduce the issue on the 4.0.0-beta.0 release but websocket proxying does not work at all. The browser side always gets a connection error with "Invalid frame header".

**Comment by alexander-akait** [comment]
@maknapp sorry for delay, maybe you can provide steps how you can reproduce it for v3?

**Comment by maknapp** [comment]
This issue was fixed in http-proxy-middleware v2.0.6.

**Pull request closed without merging**